lib/galaxy/datatypes/text.py

```diff
@@ -726,3 +727,38 @@ def sniff_prefix(self, file_prefix):
         False
         """
         return file_prefix.startswith("IQ-TREE")
+
+
+@build_sniff_from_prefix
+class Yaml(Text):
+    """Yaml files"""
+    file_ext = "yaml"
+
+    def sniff_prefix(self, file_prefix):
+        """
+            Try to load the string with the yaml module. If successful it's a yaml file.
+        """
+        return self._looks_like_yaml(file_prefix)
```

Comment:
Why what is the reason for the extra function call?

Reply:
I had been working with the json sniffer as a template. I had the same thought, but my lack of experience sold me it might be there for a good reason that I hadn't understood.